book/src/internals/build-system.md

```diff
@@ -0,0 +1,93 @@
+
+However, when linking into a static library, and then linking into the main executable, the linker will discard everything from the library that isn't used by the main executable.
+
+There are two ways to solve this:
+- Export an object file and link that to the main binary. Object files are always included completely
+- Use the whole-archive linker flag which forces inclusion of every object within the static library.
+  - If we include the entire static lib generated by cargo, then we'll likely get duplicate symbols, as this really includes **everything** that your Rust code **may** need, even if you don't use it.
+  - This has caused some recent regressions with Rust 1.78+, where MSVC could no longer link CXX-Qt due to duplicate symbols
+  - The way to solve this is to only export the static initializers as a library and link that into CMake.
+
+### Header files
+
+We want to make the generated headers available, not just to CMake, but also within dependents in the cargo build chain (e.g. your crate will probably want to depend on the headers produced by cxx-qt-lib).
+
+For this we need to export them to a stable directory so that both CMake and Cargo can find them.
+
+### (Optional) Integration with CMake
+
+Somehow, all of this should be compatible with both CMake, and Cargo-only builds.
+
+## The plan (for now)
+
+After many rounds of refactoring this, we believe the best way to go is to not rely on cargos OUT_DIR exclusivly.
+Not being able to share artifacts between crates is very limiting.
+
+We want to use a similar approach to CXX, which stores its data within either:
+- Cargos `target/` directory under the `cxxbridge` subfolder.
+- A custom `scratch` directory
+
+See: [https://github.com/dtolnay/cxx/blob/afd4aa3f3d4e5d5e9a3a41d09df3408f5f86a469/gen/build/src/target.rs#L10](https://github.com/dtolnay/cxx/blob/afd4aa3f3d4e5d5e9a3a41d09df3408f5f86a469/gen/build/src/target.rs#L10) \
+and: [https://github.com/dtolnay/cxx/blob/afd4aa3f3d4e5d5e9a3a41d09df3408f5f86a469/gen/build/src/lib.rs#L178](https://github.com/dtolnay/cxx/blob/afd4aa3f3d4e5d5e9a3a41d09df3408f5f86a469/gen/build/src/lib.rs#L178).
+
+We will likely mirror this and use the `cxxqtbridge` naming.
+
+## Contents of `cxxqtbridge`
+
+### `crates` directory
+
+Inside cxxqtbridge, there should be a `crates` folder with one subfolder per crate.
+Each crates subfolder should contain the following:
+
+- `include/`
+  - `crate-name` - A folder for all headers that are exported by this crate
+  - `cxx-qt-lib -> ../../cxx-qt-lib/include` - Symbolic links for every dependency
+- `manifest.json` - This file describes which cxx-qt-build dependencies this crate has, if it needs any Qt modules, etc.
+- `initializers.cpp` - Any code required for initialization of this crate.
+- `initializers.o` - The initializers of this crate + all it's dependencies to be linked in by CMake
```

Comment:
will qrc initialisers end up in here? as they are not part of the qml_modules but need to be somewhere?

Reply:
That will go in the qml_modules directories (see the `qml_modules directory` section)

Reply:
We could theoretically add this to the initializers.o here as well, but we likely want to keep the `cxxqt_import_qml_module` call so we'll keep this separate